titus-server-federation/src/main/java/com/netflix/titus/federation/service/FallbackJobServiceGateway.java

```diff
@@ -1,64 +1,272 @@
+        return getFallbackCompletable(methodName, primaryCompletable, secondaryCompletable);
+    }
+
+    @Override
+    public Completable updateJobStatus(JobStatusUpdate statusUpdate, CallMetadata callMetadata) {
+        final String methodName = "updateJobStatus";
+        Completable primaryCompletable = primary.updateJobStatus(statusUpdate, callMetadata);
+        Completable secondaryCompletable = secondary.updateJobStatus(statusUpdate, callMetadata);
+        return getFallbackCompletable(methodName, primaryCompletable, secondaryCompletable);
+    }
+
+    @Override
+    public Mono<Void> updateJobDisruptionBudget(JobDisruptionBudgetUpdate request, CallMetadata callMetadata) {
+        final String methodName = "updateJobDisruptionBudget";
+        Mono<Void> primaryMono = primary.updateJobDisruptionBudget(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.updateJobDisruptionBudget(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Mono<Void> updateJobAttributes(JobAttributesUpdate request, CallMetadata callMetadata) {
+        final String methodName = "updateJobAttributes";
+        Mono<Void> primaryMono = primary.updateJobAttributes(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.updateJobAttributes(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Mono<Void> deleteJobAttributes(JobAttributesDeleteRequest request, CallMetadata callMetadata) {
+        final String methodName = "deleteJobAttributes";
+        Mono<Void> primaryMono = primary.deleteJobAttributes(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.deleteJobAttributes(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Observable<Job> findJob(String jobId, CallMetadata callMetadata) {
+        final String methodName = "findJob";
+        Observable<Job> primaryObservable = primary.findJob(jobId, callMetadata);
+        Observable<Job> secondaryObservable = secondary.findJob(jobId, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
+
+    @Override
+    public Observable<JobQueryResult> findJobs(JobQuery jobQuery, CallMetadata callMetadata) {
+        final String methodName = "findJobs";
+        Observable<JobQueryResult> primaryObservable = primary.findJobs(jobQuery, callMetadata);
+        Observable<JobQueryResult> secondaryObservable = secondary.findJobs(jobQuery, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
+
+    @Override
+    public Observable<JobChangeNotification> observeJob(String jobId, CallMetadata callMetadata) {
+        // TODO: Direct stream APIs to the remote (primary) JobServiceGateway.
```

Comment:
fix the comment here? We are sending to secondary

Reply:
The TODO is to direct to the primary instead of the secondary.